Criterion/Types.hs

```diff
@@ -2,6 +2,7 @@
 {-# LANGUAGE Trustworthy #-}
```

Comment:
Similarly, we now have two different `Criterion.Types` modules across the two packages. But this time, the one in `criterion` exposes _more_ things than the one in `criterion-measurement`!

I'd favor renaming the version in `criterion-measurement` to `Criterion.Measurement.Types` to avoid this confusion. Similarly, rename `Criterion.Types.Internal` in `criterion-measurement` to `Criterion.Measurement.Types.Internal`.